tests/console/validate_lvm_raid1.pm

```diff
@@ -83,29 +77,27 @@ sub _reboot {
 }
 
 sub _check_raid_disks_after_reboot {
-    my ($config, $actual_num_devs) = @_;
+    my ($config, $expected_num_devs) = @_;
     record_info('get state after reboot');
     select_console 'root-console';
 
     assert_script_run 'mdadm --detail ' . $config->{raid1}->{name};
     my $active_devs = script_output("mdadm --detail " . $config->{raid1}->{name} . " |grep \"Active Devices\" |awk '{ print \$4 }'");
-    # if the number of disk in raid1 list are not the same report a soft failure
-    if ($active_devs != $config->{actual_num_devs}) {
-        record_soft_failure 'bsc#1150370 - disk is restored during the booting' if is_sle('>=15');
+    # if the deactivated disk is not reactivated after reboot, run _restore_raid_disk
+    unless ($active_devs == $expected_num_devs) {
+        _restore_raid_disk($config, $expected_num_devs);
     }
     else {
-        assert_equals($actual_num_devs, $active_devs, "Active devices should be still " . $config->{expected_num_devs});
+        # Expected behavior is not clear currently. Assuming that the faulty disk shouldn't be automatically reactivated
+        record_soft_failure("bsc#1172203");
     }
 }
 
 sub _restore_raid_disk {
-    my ($config) = @_;
-    my $active_devs = script_output("mdadm --detail " . $config->{raid1}->{name} . " |grep \"Active Devices\" |awk '{ print \$4 }'");
-    if ($active_devs != $config->{expected_num_devs}) {
-        assert_script_run 'mdadm --manage ' . $config->{raid1}->{name} . ' --add ' . $config->{raid1}->{disk_to_fail};
-        script_retry "mdadm --detail $config->{raid1}->{name} | grep 'Active Devices : $config->{expected_num_devs}'";
-        assert_script_run 'mdadm --detail ' . $config->{raid1}->{name};
-    }
+    my ($config, $expected_num_devs) = @_;
```

Comment:
we could use named parameters for all the functions in the file with more than one as we usually do for new test, but it was there, so minor thing if you want to change it.

Reply:
You mean instead of passing the whole $config, I could pass e.g. disk2fail => $config->{raid1}->{disk_to_fail} ,  raid_name => $config->{raid1}->{name}  ?

Reply:
I guess it's just following good practice to use hash with named parameters when it's more than one. So you write  `my %args = @_;` in the method and access `$args{config}`.
And method is called `sub(config => $config, num_dev => 3)`

But feel free to leave it for later, as you have already improved this code.

Reply:
It would help on the aspect that opening the test data wouldn't be necessary in order to understand the code, but inside the function it wouldn't make much difference.
$config->{raid1}->{disk_to_fail}  would look like $args->{disk_to_fail} 
Let's leave that for another time.

Reply:
Yes, main motivation for that is to remove any chance for error when writing parameters in the wrong order.